hw/top_earlgrey/data/placement.xdc

```diff
@@ -1,16 +1,3 @@
-############
-# Boot ROM #
-############
-# We need to enforce a defined placement of the Boot ROM to enable the Boot ROM splicing script.
-# Any change in ROM instances path should be updated in following two files
-# 1. hw/top_earlgrey/data/placement.xdc and
-# 2. hw/top_earlgrey/util/vivado_hook_opt_design_post.tcl
-set_property LOC RAMB36_X0Y10 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_0" && PRIMITIVE_TYPE =~ BMEM.*.* }]
-set_property LOC RAMB36_X0Y11 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_1" && PRIMITIVE_TYPE =~ BMEM.*.* }]
-set_property LOC RAMB36_X0Y12 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_2" && PRIMITIVE_TYPE =~ BMEM.*.* }]
-set_property LOC RAMB36_X0Y13 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_3" && PRIMITIVE_TYPE =~ BMEM.*.* }]
-set_property LOC RAMB36_X0Y14 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_4" && PRIMITIVE_TYPE =~ BMEM.*.* }]
-
```

Comment:
sorry i'm a bit late to the party, but could someone quickly explain for a minute why we don't need the boot rom placement anymore? I thought we needed to find these in order to know where to splice.

Reply:
The thing is: the nice script you wrote some time ago actually extracts the location of the BRAMs in the design. If the script is called late enough (meaning after place and route) these locations are indeed known and there is no point in setting a placement constraint. All that is needed is to verify that the ROM is indeed implemented using BRAM such that your script can extract the locations. This check is performed as part of the opt_design post hook.

Reply:
The advantage of not specifying the locations is that the tool is free to place the ROM where suitable. On the CW310, I noted that the ROM logic was usually placed pretty far away from the ROM BRAMs. Also, without LOC constraints there won't be issues e.g. because on a different FPGA, the specified locations do not exist.